### DISCUSSION TITLE
Using skin elements for streaming overlays

### DISCUSSION THREAD

**Discussion by Chiffario** [ppy/osu#16949] [Ideas] [answered]
Currently osu! streamers on platforms like Twitch and Youtube use memory reading based tools like [BlackShark's gosumemory](https://github.com/l3lackShark/gosumemory) and [Piotrekol's StreamCompanion](https://github.com/Piotrekol/StreamCompanion) to display different (mostly additional) gameplay elements in their [overlays](https://camo.githubusercontent.com/6d51d3d239e2a945d9d324ae4f8dd577e4f2efcdcf075efc8668949a1f8a68db/68747470733a2f2f692e696d6775722e636f6d2f6e32773236306f2e6a7067). 

The main idea behind this suggestion is that creating a built-in overlay support would fix a lot of issues with "outside" methods while also making them more user-friendly.

My idea is to create a system which would allow to create overlay parts from skin elements, maybe by creating additional transparent windows which could be window-captured by OBS or (iirc) via WebSockets as aforementioned open-source overlays do. It would let players combine personally comfortable layouts (hiding certain skin elements f.i. combo in actual gameplay to avoid nerves or screen cluttering) with viewer-focused ones (showing certain elements in the capture overlay f.i. combo counter on stream only to let viewers see more information)

I'm aware that it won't be implemented for a good while if ever, but I thought it is worth mentioning here since it could be a great addition for players in the future

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/discussions/15151
https://github.com/ppy/osu/discussions/14945
https://github.com/ppy/osu/discussions/15061

**Reply by Wieku** [reply]
~Also #12274 which fixes all problems gosumemory and StreamCompanion have with stable.~
It's linked in linked discussions
